src/services/module_service.ts

```diff
@@ -41,115 +44,133 @@ export default class ModuleService {
+        const data = await ModuleService.constructDataForDenoImport(dep);
+        allModules.push(data);
       }
+    }
 
-      // Get the repository url (always Github for deno.land but could be something else for nest.land)
-      const repositoryURL: string = isStd
-        ? "https://github.com/denoland/deno/tree/master/std/" + name
-        : await RegistryService.getThirdPartyRepoURL(name);
-
-      // Get the latest release - make sure the string is the same format as imported version eg using a "v"
-      const latestRelease: string = isStd
-        ? ModuleService.standardiseVersion(
-          importedVersion,
-          await RegistryService.getLatestModuleRelease("std"),
-        )
-        : ModuleService.standardiseVersion(
-          importedVersion,
-          await RegistryService.getLatestModuleRelease(name),
-        );
+    return allModules;
+  }
 
-      // Get the description
-      const description: string = !isStd
-        ? await RegistryService.getThirdPartyDescription(name)
-        : colours.red(
-          "Descriptions for std modules are not currently supported",
-        );
+  private static getImportedVersionFromImportLine(importLine: string): string {
+    const importVersionRegexResult = importLine.match(importVersionRegex);
+    const importedVersion: string = importVersionRegexResult !== null &&
+        importVersionRegexResult.length > 0
+      ? importVersionRegexResult[0]
+      : "";
+    return importedVersion;
+  }
 
-      // Save the module
-      modules.push({
-        std: isStd,
-        repositoryURL,
-        moduleURL,
-        latestRelease,
-        importedVersion,
-        name,
-        description,
-      });
+  private static async constructDataForNestImport(
+    importLine: string,
+  ): Promise<IModule> {
+    const isStd = importLine.indexOf("/std") > -1;
+    const importUrl: string = importLine.substring(
+      importLine.indexOf("https://"),
+      importLine.indexOf(".ts") + 3, // to include the `.ts`
+    );
+    const importedVersion = ModuleService.getImportedVersionFromImportLine(
+      importLine,
+    );
+    let name = "";
+    if (isStd) {
+      const nameAndFile = importLine.split("@" + importedVersion + "/")[1];
+      name = nameAndFile.split("/")[0];
+    } else {
+      const nameAndVersionAndFile = importLine.split("https://x.nest.land/")[1];
+      name = nameAndVersionAndFile.split("@")[0];
     }
+    const repositoryUrl = await NestService.getThirdPartyRepoURL(name);
+    const latestRelease = ModuleService.standardiseVersion(
+      importedVersion,
+      await NestService.getLatestModuleRelease(name),
+    );
+    const description = await NestService.getThirdPartyDescription(name);
+    return {
+      description,
+      latestRelease,
+      repositoryUrl,
+      name,
+      std: isStd,
+      importedVersion,
+      importUrl,
+    };
+  }
 
-    if (!modules.length) {
-      return false;
+  private static async constructDataForDenoImport(
```

Comment:
this method has some of the same logic as `constructDataForNestImport()`. any reason we make a function with the logic that's the same? or do we not want to do that because these methods might change in the future based on a module being on nest vs deno?

Reply:
there are a few subtle differences, the only similarities are the variable names - the values assigned to those variables are already using reusable methods

Reason for these different methods is because the methods do logic specific to the registry, granted there are some duplications, but its just things like `val = str.indexOf(...)`

Reply:
ah, ok. LGTM